src/main/java/com/mineboundteam/minebound/magic/MagicEvents.java

```diff
@@ -236,4 +239,15 @@ protected static NonNullConsumer<PlayerMana> handlePlayerMana(Player player) {
             CapabilitySync.NET_CHANNEL.send(PacketDistributor.PLAYER.with(() -> (ServerPlayer) player), new CapabilitySync.ManaSync(mana.getMana(), totalMana));
         };
     }
-}
+
+    @SubscribeEvent
+    public static void onPlayerCloned(PlayerEvent.Clone event){
+        if(event.isWasDeath() && ManaConfig.keepArmor.get()){
+            for(ItemStack item : event.getOriginal().getArmorSlots()){
+                if(!item.isEmpty() && item.getItem() instanceof MyrialArmorItem){
+                    event.getPlayer().setItemSlot(Player.getEquipmentSlotForItem(item), item);
+                }
+            }
+        }
+    }
```

Comment:
Isn't this also in `MyrialArmorItem`? Should remove it from there.

Reply:
Fixed; the code actually needed to be significantly expanded to work; should be fixed now